javascript/ql/src/semmle/javascript/dataflow/TaintTracking.qll

```diff
@@ -623,6 +627,13 @@ module TaintTracking {
     }
 
     override predicate sanitizes(boolean outcome, Expr e) {
+      // Explanation for why `if(~whitelist.indexOf(x))` is equivalent to `if(whitelist.indexOf(x) != -1)`:
```

Comment:
I'm not sure this is needed, at least not at this level of detail. Just reminding the reader that `~x` is 0 iff `x` is -1 should be fine.

Reply:
I have put the simplified explanation in the qldoc.